cfpb/college-costs
# Display school and national data in bar graphs

Bar graphs that show real data (and handle all the edge cases that come with showing real data)
## Additions
- `data` attributes in bar graph HTML to set graph parameters
- `metric-view` to handle setting all the graphs
- Unit tests
- Functional tests
- Top label ("$100,000") in the salary graph
## Changes
- "Your school" graph labels now say "This school"
## Testing
- To test, pull in the `bar-graphs` branch and run `gulp`. Fire up the app in standalone or UnityBox as normal.
- Try testing with our usual example school, good old `iped=408039`. The graphs should look like the screen shots below.
- Try testing with any other schools you like. The graphs should show the values in `window.schoolData` and `window.nationalData`. A particularly good outlier is `iped=239169`: it has no reported graduation rate, a 0% default rate, and a $250,000 average salary! In that example, the school point in the graduation graph and the "$100,000" label in the salary graph should both be gone.
- If you test other schools, note that this PR **does not** include getting the alerts below each graph working, so don't worry about those for now.
- If anyone knows of a school where the school's average is exactly the same as the national average, that graph should only show the school point.
- I've tested this in IE8/Win7, IE9/Win7, IE10/Win7, IE11/Win7, Firefox/Win7, Chrome/Mac, Safari/Mac, Mobile Safari/iPhone6. Everything seems to be looking good except for IE8, where an unrelated JS error is still preventing an scripts from running.
## Review
- @marteki 
- @mistergone: How's all the JS and testing look?
- @higs4281 
- @ascott1: Totally optional, but if you want to give the JS and testing another look, let me know if you see anything that needs improvement.
## Screenshots

(Not all the screen sizes, but you get the idea)

| 600px and below | 1230 px and above |
| --- | --- |
| ![graphs-small](https://cloud.githubusercontent.com/assets/1862695/13132376/170c35b6-d5c0-11e5-8549-366eb85e0f97.png) | ![graphs-large](https://cloud.githubusercontent.com/assets/1862695/13132378/1b3df188-d5c0-11e5-9ac8-78eaeae7ba2e.png) |
## Todos
- Add functional test for edge case schools
- Change "This school" labels to "This program" in graphs where program data is used
- Update project documentation
## Checklist
